### PR TITLE
Support For Dynamic Property Update & Linear Gradient

### PR DESCRIPTION
**Pull request by shefayaparwin**
# Summary

Support For Dynamic Property Update & Linear Gradient & implemented few missed properties

## Major Change List
- [x]  Added support for dynamic property Update
- [x]  Supported Linear Gradient SVG Element
- [x]  With Linear Gradient SVG Element support, enabled fill & stroke to use color type: brushRef/URI

## Supported List of other missed properties
- [x]  Added support to missed out transform property "Rotate"
- [x] Added Support to Apply Property specified on Use element on Def Nodes

## Issues Fixed
- [x] Removed temporary Fix made to match default color behavior as transparent as of RN iOS & Android. With which default color is black, which is similar to RN web & svg CSS

## Minor changes
- [x]  Changes made to ease out Debug by wrapping debug under common macro and made debugs for meaning
- [x]  Removed unnecessary comment.
- [x] Removed unsed APIs for property conversion.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Linux|    ✅     |
| Ubuntu|    ✅     |
| iOS     |    ❌     |
| Android |    ❌     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added documentation in `README.md`
- [x] I added a test for the API in the `__tests__` folder

### CODE REVIEW

File: skia/RSkSVGComponentNode.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  if( rootSvgNode && (rootSvgNode->tag() == SkSVGTag::kSvg)) {
+    auto node=dynamic_cast<RSkComponentRNSVGSvgView *>(rootSvgNode);
+    if(node) {
+      return node->getContainerSize();
+    }
+  }
+  return SkSize::Make(0, 0);
+}
+
+void RSkSVGComponentNode::invalidateLayer(){
```
Comment (munezbn) [inline skia/RSkSVGComponentNode.cpp]: rename it to invalidateParentSvgContainer()

File: skia/RSkComponentRNSVGLinearGradient.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      if (const auto* y2 = attrValue.as<SkSVGLengthValue>()) {
+        y2_ =*y2;
+      }
+      break;
+    default:
+      this->INHERITED::onSetAttribute(attr, attrValue);
+  }
+}
+
+sk_sp<SkShader> RSkComponentRNSVGLinearGradient::getShader(const SkSVGRenderContext& ctx)  {
```
Comment (munezbn) [inline skia/RSkComponentRNSVGLinearGradient.cpp:116]: Handle invalid brush negative case

File: skia/RSkSVGContainer.cpp [hunk 3]
```diff
@@ -107,32 +95,67 @@ bool RSkSVGContainer::hasChildren() const {
   return !childRSkNodeList_.empty();
 }
 
+bool RSkSVGContainer::onPrepareToRender(SkSVGRenderContext* ctx) const {
+  /*For  colorType- uri/BRUSH_REF, inheritence  to be handled in RSKSVG*/
+  auto inheritColor =[&](RSkSVGNode* childNode, std::string colorType,RNSVGColorFillStruct colorStruct){
+
+    if((std::find (childNode->renderablePropList.begin(), childNode->renderablePropList.end(), colorType) == childNode->renderablePropList.end())) {
+
+      if((std::find (renderablePropList.begin(), renderablePropList.end(), colorType) != renderablePropList.end())) {
```
Comment (munezbn) [inline skia/RSkSVGContainer.cpp:103]: This should be first if check.

File: skia/RSkSVGNode.cpp [hunk 4]
```diff
@@ -339,39 +291,66 @@ bool RSkSVGNode::setDashArrayAttribute(SkSVGAttribute attr, const std::vector<st
...
                   " Matrix 3 : "<<matrix[3] << "\n" <<
                   " Matrix 4 : "<<matrix[4] << "\n" <<
                   " Matrix 5 : "<<matrix[5]);
+#endif/*ENABLE_RSKSVG_PROPS_DEBUG*/
 
-    SkMatrix svgTransforMatrix=SkMatrix::Translate(matrix[4],matrix[5]);
-    svgTransforMatrix.preConcat(SkMatrix::Scale(matrix[0],matrix[3]));
-    setAttribute(attr,SkSVGTransformValue(svgTransforMatrix));
+    SkMatrix svgTransforMatrix;
+    // Converting received Matrix from React Native framework which is in column major Order to row Major Order for SKia
```
Comment (munezbn) [inline skia/RSkSVGNode.cpp:306]: Can this done with any copy constructor

File: skia/RSkComponentRNSVGLinearGradient.cpp (hunk 2 above)
Comment (shefayaparwin) [reply skia/RSkComponentRNSVGLinearGradient.cpp:116]: Right now, Invalid Brush Handled while calling get shader, by ensuring component is LinearGradient

File: skia/RSkSVGNode.cpp (hunk 4 above)
Comment (shefayaparwin) [reply skia/RSkSVGNode.cpp:306]: SkMatrix doesn't have copy constructor, which takes column major & converts to row major

**Review by munezbn — APPROVED** [review]

**Review by chatrapathi06 — APPROVED** [review]

**Pull request merged**